File: solutions/剑指 Offer 06. 从尾到头打印链表/solution1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> reversePrint(ListNode* head) {
        vector<int> res;
        stack<int> stk;
        while (head) {
            stk.push(head->val);
            head = head->next;
        }
        while (!stk.empty()) {
            res.push_back(stk.top());
            stk.pop();
        }
        return res;
    }
};

File: solutions/剑指 Offer 06. 从尾到头打印链表/solution1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        stk = []
        while head:
            stk = [head.val] + stk
            head = head.next
        return stk
